iphone/Maps/Classes/CustomViews/NavigationDashboard/MWMNavigationDashboardManager.mm

```diff
@@ -172,17 +172,26 @@ - (void)stateError {
 
 - (void)stateReady {
   // TODO: Here assert sometimes fires with _state = MWMNavigationDashboardStateReady, if app was stopped while navigating and then restarted.
+  // Also in ruler mode when new point is added by single tap on the map state MWMNavigationDashboardStatePlanning is skipped and we get _state = MWMNavigationDashboardStateReady.
+  if (_state == MWMNavigationDashboardStateReady)
```

Comment:
With single-tap ruler we get `_state == MWMNavigationDashboardStateReady` on each tap.
Should we comment out `NSAssert(...)` or remove this `if (...)`.

Reply:
Why is this stateReady needed for each nav type? Logically, we can later add "Start navigation" to the ruler too.

The question is why the assert below happens (why is it called with MWMNavigationDashboardStatePlanning)?

Guessing is not the best strategy here, it may lead to many unexpected bugs...

Reply:
When we build or update route from iOS UI method `MWMRouter rebuildWithBestRouter: NO` is invoked which changes state to `MWMNavigationDashboardStatePlanning` and calls `RoutingManager.BuildRoute()`.

But with current implementation `RoutingManager.BuildRoute()` is invoked not from UI code but from `Framework::OnTapEvent()` and it doesn't trigger planning state.